pkg/bazel.go

```diff
@@ -0,0 +1,44 @@
+package pkg
+
+import (
+	"io"
+	"os/exec"
+)
+
+type BazelCmdConfig struct {
+	// Dir represents the working directory to use for the command.
+	// If Dir is the empty string, use the calling process' current directory.
+	Dir string
+
+	// Stdout and Stderr specify the process's standard output and error.
+	// A nil value redirects the output to /dev/null.
+	// The behavior is the same as the exec.Command struct.
+	Stdout io.Writer
+	Stderr io.Writer
+}
+
+type BazelCmd interface {
+	Run(config BazelCmdConfig, args ...string) (int, error)
```

Comment:
Possibly I'm being overly cautious here, but how would you feel about naming this `Exec` or `Execute` instead of `Run`? I can imagine reading `bazel.Run` as if it does a `bazel run` (and having `BazelCmd.{Build,Test,Run}(...)` rather than `BazelCmd.Execute("build", ...)` would be a completely reasonable alternate API choice)

Reply:
Good point, I'll change that.